node/primitives/src/approval.rs

```diff
@@ -121,3 +133,84 @@ pub struct BlockApprovalMeta {
 	/// The consensus slot of the block.
 	pub slot: Slot,
 }
+
+/// Errors that can occur during the approvals protocol.
+#[derive(Debug, thiserror::Error)]
+#[allow(missing_docs)]
+pub enum ApprovalError {
+	#[error("Schnorrkel signature error")]
+	SchnorrkelSignature(schnorrkel::errors::SignatureError),
+	#[error("Authority index {0} out of bounds")]
+	AuthorityOutOfBounds(usize),
+}
+
+/// An unsafe VRF output. Provide BABE Epoch info to create a `RelayVRFStory`.
+pub struct UnsafeVRFOutput {
+	vrf_output: VRFOutput,
+	slot: Slot,
+	authority_index: u32,
+}
```

Comment:
I'd say `UnsafeVRFPreOutput` or maybe `UncheckedVRFPreOut`

Reply:
sounds better, but i'm not gonna dig into `substrate-consensus-vrf` to make the name change just for this. rename should happen in schnorrkel and then propagate via Substrate

Reply:
I'll try to do this soon.